app/Http/Controllers/Admin/BookController.php

```diff
@@ -25,6 +29,71 @@ public function create()
         return view('backend.books.create', compact('categories'));
     }
 
+    /**
+     * Store a newly book created resource in storage.
+     *
+     * @param App\Http\Requests\BookCreateRequest $request get create request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(BookCreateRequest $request)
+    {
+        $book = new Book($request->toArray());
+        // save image path, move image to directory
+        if (isset($request->image)) {
+            $image = $request->image;
+            $name = config('image.name_prefix') . "-" . $image->hashName();
+            $folder = config('image.books.path_upload');
+            $saveImageResult = $image->move($folder, $name);
+
+            $book->image = $name;
+        } else {
+            $book->image = config('image.books.no_image_name');
+            $saveImageResult = true;
+        }
+
+        //save new donator
+        $user = User::where('employee_code', $request->employee_code)->first();
+        if ($user === null) {
+            $donatorData = [
+                'employee_code' => $request->employee_code,
+            ];
+        } else {
+            $donatorData = [
+                'user_id' => $user->id,
+                'employee_code' => $user->employee_code,
+                'email' => $user->email,
+                'name' => $user->name,
+            ];
+        }
+        $donator = Donator::updateOrCreate(['employee_code' => $request->employee_code], $donatorData);
```

Comment:
- I think you should check updateOrCrate function. if you use this function, user donate only one book. please check it.